### ISSUE TITLE
[BUG] rna2vec and encode_rna return 1D shape (0,) on empty input, violating documented (n, max_len) contract

### ISSUE THREAD

**Issue by officialasishkumar** [gc-os-ai/pyaptamer#564]
LLM generated content, by Claude (Opus 4.7)

**Describe the bug**

Both `pyaptamer.utils.rna2vec` and `pyaptamer.utils.encode_rna` document a 2D output shape of `(n_sequences, max_sequence_length)` (resp. `(n_sequences, max_len)`), but when the input list is empty (or every sequence is silently dropped, e.g. all-unknown triplets in `rna2vec`), they fall through `np.array([])` and return a 1D array of shape `(0,)` instead. This violates the documented shape contract: any caller that does `result.shape[1]`, slices `result[:, k]`, stacks the result with another `(n, max_len)` array, or feeds it into a `nn.Embedding` will fail.

**To Reproduce**

```python
from pyaptamer.utils import encode_rna, rna2vec

# rna2vec on an empty input
print(rna2vec([], sequence_type="rna", max_sequence_length=275).shape)
# observed: (0,)
# expected: (0, 275)

# encode_rna on an empty input (PyTorch tensor)
print(encode_rna([], words={"A": 1}, max_len=10).shape)
# observed: torch.Size([0])
# expected: torch.Size([0, 10])

# encode_rna on an empty input (numpy)
print(encode_rna([], words={"A": 1}, max_len=10, return_type="numpy").shape)
# observed: (0,)
# expected: (0, 10)
```

**Expected behavior**

For an empty input list, both functions should return an array/tensor whose shape matches the documented `(n, max_len)` contract: zero rows, but `max_len` (or `max_sequence_length`) columns.

This makes the functions safe to call on empty batches and lets downstream code chain calls (e.g. `np.concatenate`, `torch.stack`, slicing along axis 1) without conditional shape patching at every site.

**Additional context**

- `rna2vec` already builds a list of padded rows and only collapses to a 0-D shape when `result` is `[]`. The fix is one extra branch that constructs an array of shape `(0, max_sequence_length)` with the right dtype.
- `encode_rna` already pads to `max_len` per row, so the fix is to skip building an empty array and instead return `np.empty((0, max_len), dtype=np.int64)` (then wrap in a `torch.tensor` for the default path).
- Both functions internally also drop sequences that produce no triplets / no matched tokens, so this bug is also reachable in practice when a user passes a batch consisting entirely of too-short or all-unknown strings (see related issues #511, #372, #363, #284 for the silent-drop behaviour, but the shape contract issue is independent).

**Versions**

```
pyaptamer 0.1.0a1 (main, commit fd41ee6)
```
